### DISCUSSION TITLE
Incentives for chain operators

### DISCUSSION THREAD

**Discussion by vayesy** [playzero/zero-network#181] [General]
As a parachain we would like to implement rewards for the users, participating in our chain infrastructure and making it more decentralized. 
The goal of this discussion is to come up with concrete list of incentive programs we would implement on our chain.

I can see few options we can choose from:
1. Introduce premium to collator authoring the block for finalization. This is similar to validator rewards but we would need to think of the source of those rewards. It can be pool of existing tokens, which will be used to share rewards from or new tokens can be minted on the fly to give reward from, implies inflation.
2. NFT rewards. Purpose of those NFTs could vary:
   * exchange NFTs later for tokens;
   * give permissions to do some actions on chain;
   * add respectable labels to account;
   * etc.

Feel free to add other incentive options we can use.

@2075 ([source](https://github.com/playzero/zero-network/pull/167#issuecomment-1286838837)):

> - alphaville was a npos prototype incentivising validator operators to contribute to building a decentralised network. for me the way to go, also according to our product thesis.
> - subzero as a parachain lacks incentive for people to operate at the moment.
> - the way it is built now assumes a permissioned consortium style network, which is not where we are heading at or from my point of view could but should not head to.
> - as a result we should incentivise people to
> - operate collators
> - get rewards for that
> - have a determined amount of collators which can be nominated / delegated
> - from all collators a subset are selected e.g. random to be in the collator set
> - therefore incentive is there to run a collator and through nomination/delegation/selection enough incentive is there to support as a community and get commercial rewards

@vovacha @FiberMan @5-mark

**Comment by vovacha** [comment]
I'd like to understand how other parachains approach this problem. @vayesy, do we have this information?

**Comment by 2075** [comment]
- i think we should not reinvent the wheel on the last mile
- just use existing pallets which are also utilised in other parachains
- the rewards will be payed through the tx fees usually

**Reply by FiberMan** [reply]
Polkadot has quite high requirements for collator hardware. I think it's relevant for Zero also.
Thus, rewards must be profitable enough.

But as we are in the game industry, maybe some PRO gamers may create collators for their own good.
E.g. guilds in GameDAO may have a collator(s) which may bring them more loyalty points from web3 fans. 
We can introduce an appropriate badge for that visible to the community.

**Reply by 2075** [reply]
good thoughts.

yes, the requirements of a polkadot node are similar to a zero node and should be considered. pro node operators are aware of these requirements and setup industry standard HA clusters etc to run nodes reliably.

a collator or validator should not be run on premise on your gaming rig, at least not until we can properly create nodes which are suitable for such a setup. such a setup could be the "compute" part of zero network, where we can e.g. reward compute contributors based on their computed performance of ad hoc or continuously provisioned hardware.

we can for sure incentivise daos to run collators, but should come up with a good plan from initial more centralised setup to onboard continuously community collators.
